### ISSUE TITLE
Can't publish messages to multiplexed channel when using redis

### ISSUE THREAD

**Issue by argb** [slact/nchan#276]
```
curl --request POST --data "test message" http://127.0.0.1:8001/pubsub/dd
curl: (52) Empty reply from server
```
Assertion failed: (!cf->redis.enabled), function nchan_store_chanhead_publish_message_generic, file /usr/local/share/nchan-nginx-module/src/store/memory/memstore.c, line 2778.
2016/11/16 15:51:59 [alert] 6630#0: worker process 6633 exited on signal 6

The config:
```nginx
location ~ /pubsub/(\w+)$ {
        nchan_channel_id "$1" "common_channel";
        nchan_pubsub;
        nchan_use_redis on;
        nchan_redis_url 127.0.0.1:6379;

        #nchan_redis_pass redis_cluster;
    }
```
At beginning, I try to use redis cluster with one redis server, but it failed with the same error, then I changed it like above, it doesn't work yet.

The OS envirement is MacOS Sierra.

By the way, could you provide a full featured configure example?

**Comment by slact** [comment]
Ah, this is a bug triggered by redis + publishing to multiple channels. I will have it fixed shortly, but in the meantime, I doubt this is what you actually wanted to do. I assume you wanted to publish only to channel $1 and not to "common channel". This can be done in a pubsub location by having a separate `nchan_publisher_channel_id` and `nchan_subscriber_channel_id`:

```nginx
location ~ /pubsub/(\w+)$ {
        nchan_publisher_channel_id "$1";
        nchan_subscriber_channel_id "$1" "common_channel";
        nchan_pubsub;
        nchan_use_redis on;
        nchan_redis_url 127.0.0.1:6379;

        #nchan_redis_pass redis_cluster;
}
```

I will have this bug fixed shortly.

**Comment by slact** [comment]
Fixed in 1256628. Please let me know if the issue persists for you.

As for a full-featured example, there are several littered throughout the documentation. I will be adding examples for some common use cases in the future, but because the configuration is very flexible, I can't guarantee that those would be definitive and full-featured. Still, it might be good to have them in one place.
